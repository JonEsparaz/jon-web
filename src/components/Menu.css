@media (min-width: 768px) {
}

@media (max-width: 767px) {

    .navbar-toggler {
        outline-color: #ffffff00 !important;
        border-color: #ffffff00 !important;
        background-color: #ffffff00 !important;
        left:86.5vw;
        top:5.35vw;
        height:8vw;
    }
}