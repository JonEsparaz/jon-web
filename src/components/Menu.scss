@import '../theme.scss';

.navlink-custom {
    text-decoration: none !important;
}

.navlink-custom:hover {
    color: inherit;

    .Underline:before, .Underline2:before {
        visibility: visible;
        transform: scaleX(1);
    }
}

.white-link {
    color: white !important;
}

@media (min-width: 768px) {
    .MenuContainer {
        margin-left: 7.5vw;
        width: 80vw;
    }

    .navlink-custom {
        color: black;
        font-family: $font;
        text-transform: uppercase;
        font-size: 14px;
        margin-right: 1rem;
    }
}

@media (max-width: 767px) {
    .MenuContainer {
        width: 100vw;
    }

    .navbar-toggler {
        outline: none;
        background-color: #ffffff00 !important;
        left:86.5vw;
        top:5.35vw;
        height:8vw;
    }

    .navlink-custom {
        color: black;
        font-family: $font;
        text-transform: uppercase;
        font-size: 16px;
    }
}