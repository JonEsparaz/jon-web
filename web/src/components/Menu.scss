@use './hamburgers.scss';

.light-link {
  color: white !important;
}

.dark-link {
  color: black !important;
}

.logo {
  height: 80px;
  margin-right: 2vw;
}

.menu-icon {
  margin-right: 5px;
  margin-bottom: 2px;
}

.active-link {
  .underline-dark:before,
  .underline-light:before {
    visibility: visible;
    transform: scaleX(1);
  }
}
@media (hover: hover) {
  .navlink-custom:hover {
    color: inherit;

    .underline-dark:before,
    .underline-light:before {
      visibility: visible;
      transform: scaleX(1);
    }
  }
}

@media (min-width: 768px) {
  .navlink-custom {
    font-size: 14px;
    margin-right: 1rem;
  }
}

@media (max-width: 767px) {
  .navbar-toggler {
    outline: none;
    background-color: #ffffff00 !important;
    left: 86.5vw;
    top: 5.35vw;
    height: 8vw;
  }

  .navlink-custom {
    font-size: 16px;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .active-link {
    text-decoration: underline !important;
    text-underline-offset: 2px;
  }
}
