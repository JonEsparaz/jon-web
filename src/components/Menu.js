import React from 'react';
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';
import HamburgerMenu from 'react-hamburger-menu';
import AppsIcon from '@material-ui/icons/Apps';
import CodeIcon from '@material-ui/icons/Code';
import EmailIcon from '@material-ui/icons/Email';
import './Menu.css';

export default class Menu extends React.Component {

  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = { isOpen: false };
  }

  toggle() {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  render() {
    return (
      <div className="MenuContainer">
        <Navbar light expand="md">
        <NavbarBrand href="/"><img src="/mylogo.png" style={{height: "80px", marginRight: "0.5vw", marginLeft: "3vw", marginTop: "2.5vw", marginBottom: "2.5vw"}}></img></NavbarBrand>
          <NavbarToggler onClick={this.toggle}>
            <HamburgerMenu isOpen={this.state.isOpen} menuClicked={this.toggle.bind(this)} width={24} height={16} strokeWidth={2} borderRadius={45} color="black"/>
          </NavbarToggler>
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav navbar>
            <NavItem>
              <NavLink href="/speedcubing"><AppsIcon style={{marginRight: "5px"}}/>Speedcubing</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/projects"><CodeIcon style={{marginRight: "5px"}}/>Projects</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/contact"><EmailIcon style={{marginRight: "5px"}}/>Contact</NavLink>
            </NavItem>            
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}