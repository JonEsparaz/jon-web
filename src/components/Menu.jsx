import React from 'react';
import AppsIcon from '@material-ui/icons/Apps';
import CodeIcon from '@material-ui/icons/Code';
import EmailIcon from '@material-ui/icons/Email';
import { Collapse, NavbarToggler, Nav, NavLink, Navbar, NavbarBrand } from 'reactstrap';
import HamburgerMenu from 'react-hamburger-menu';
import './Menu.scss';

export default class Menu extends React.Component {

  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = { isOpen: false };
  }

  toggle() {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  render() {
    return (
      <div className="MenuContainer">
        <Navbar expand="md">
          <NavbarBrand href="/"><img alt="logo" src="/white-mylogo.png" style={{height: "80px", marginRight: "0.5vw"}}></img></NavbarBrand>
          <NavbarToggler onClick={this.toggle}>
            <HamburgerMenu isOpen={this.state.isOpen} menuClicked={this.toggle} width={24} height={16} strokeWidth={2} borderRadius={45} color="white" />
          </NavbarToggler>
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav navbar className="mr-auto" >
              <NavLink className="navlink-custom" href="/speedcubing"><AppsIcon style={{marginRight: "5px", marginBottom: "2px"}} />Speedcubing</NavLink>
              <NavLink className="navlink-custom" href="/projects"><CodeIcon style={{marginRight: "5px", marginBottom: "2px"}}/>Projects</NavLink>
              <NavLink className="navlink-custom" href="/contact"><EmailIcon style={{marginRight: "5px", marginBottom: "2px"}}/>Contact</NavLink>
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}