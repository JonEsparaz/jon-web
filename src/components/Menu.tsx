import React, { useState } from 'react';
import { Collapse, NavbarToggler, Nav, Navbar } from 'reactstrap';
import { NavLink, Link } from 'react-router-dom';
import './Menu.scss';

export interface Props {
  mode: 'light' | 'dark';
  absolute?: boolean;
}

export default function Menu({ mode, absolute }: Props): JSX.Element {
  const [isOpen, setIsOpen] = useState(false);

  const links = [
    {
      to: '/speedcubing',
      title: 'Speedcubing',
      icon: `/svg/cubing-${mode}.svg`,
      alt: '',
    },
    {
      to: '/contact',
      title: 'Contact',
      icon: `/svg/email-${mode}.svg`,
      alt: '',
    },
  ];

  return (
    <div
      className={`px-4 px-sm-5 w-100 ${
        absolute ? 'position-absolute' : 'position-relative'
      }`}
      style={{
        zIndex: absolute ? 9999 : undefined,
      }}
    >
      <Navbar expand="md">
        <Link to="/">
          <img alt="logo" src={`/logos/je-logo-${mode}.png`} className="logo" />
        </Link>
        <NavbarToggler onClick={() => setIsOpen(!isOpen)}>
          <div
            className={`hamburger hamburger--slider ${mode}-burger ${
              isOpen ? 'is-active' : ''
            }`}
          >
            <div className="hamburger-box">
              <div className="hamburger-inner" />
            </div>
          </div>
        </NavbarToggler>
        <Collapse isOpen={isOpen} navbar>
          <Nav navbar className="mr-auto">
            {links.map((link) => {
              return (
                <li key={link.title}>
                  <NavLink
                    className={`navlink-custom text-decoration-none text-uppercase ${
                      mode === 'light' ? 'text-white' : ''
                    }`}
                    activeClassName="active-link"
                    to={link.to}
                  >
                    <img
                      src={link.icon}
                      alt={link.alt}
                      className="menu-icon"
                      width={24}
                      height={24}
                    />
                    <span className={`underline-${mode}`}>{link.title}</span>
                  </NavLink>
                </li>
              );
            })}
          </Nav>
        </Collapse>
      </Navbar>
    </div>
  );
}
