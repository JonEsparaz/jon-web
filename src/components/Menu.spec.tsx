import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import Menu from './Menu';

describe('navigation menu', () => {
  test('snapshots, light/dark mode, absolute/relative position', () => {
    const history = createMemoryHistory();
    const { container, rerender } = render(
      <Router history={history}>
        <Menu mode="light" />
      </Router>,
    );

    expect(container.innerHTML).toMatchSnapshot();

    rerender(
      <Router history={history}>
        <Menu mode="light" absolute />
      </Router>,
    );

    expect(container.innerHTML).toMatchSnapshot();

    rerender(
      <Router history={history}>
        <Menu mode="dark" />
      </Router>,
    );

    expect(container.innerHTML).toMatchSnapshot();

    rerender(
      <Router history={history}>
        <Menu mode="dark" absolute />
      </Router>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('navbar toggler', async () => {
    Object.assign(window, { innerWidth: 300 });

    const history = createMemoryHistory();
    const { getByLabelText, container } = render(
      <Router history={history}>
        <Menu mode="light" />
      </Router>,
    );

    // toggler button
    const toggler = getByLabelText('Toggle navigation');

    // navbar is collapsed
    expect(container.getElementsByClassName('show').length).toBe(0);

    // hamburger component is closed
    expect(toggler.innerHTML).toMatchSnapshot();

    // click button
    fireEvent.click(toggler);

    // navbar is open
    await waitFor(() =>
      expect(container.getElementsByClassName('show').length).toBe(1),
    );

    // hamburger component is open
    expect(toggler.innerHTML).toMatchSnapshot();
  });
});
