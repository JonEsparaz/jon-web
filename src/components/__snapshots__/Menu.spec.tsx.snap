// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation menu navbar toggler 1`] = `"<div class=\\"hamburger hamburger--slider light-burger \\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div>"`;

exports[`navigation menu navbar toggler 2`] = `"<div class=\\"hamburger hamburger--slider light-burger is-active\\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div>"`;

exports[`navigation menu snapshots, light/dark mode, absolute/relative position 1`] = `"<div class=\\"px-4 px-sm-5 w-100 position-relative\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo-light.png\\" class=\\"logo\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider light-burger \\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><li><a class=\\"navlink-custom text-decoration-none text-uppercase text-white\\" href=\\"/speedcubing\\"><img src=\\"/svg/cubing-light.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-light\\">Speedcubing</span></a></li><li><a class=\\"navlink-custom text-decoration-none text-uppercase text-white\\" href=\\"/contact\\"><img src=\\"/svg/email-light.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-light\\">Contact</span></a></li></ul></div></nav></div>"`;

exports[`navigation menu snapshots, light/dark mode, absolute/relative position 2`] = `"<div class=\\"px-4 px-sm-5 w-100 position-absolute\\" style=\\"z-index: 9999;\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo-light.png\\" class=\\"logo\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider light-burger \\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><li><a class=\\"navlink-custom text-decoration-none text-uppercase text-white\\" href=\\"/speedcubing\\"><img src=\\"/svg/cubing-light.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-light\\">Speedcubing</span></a></li><li><a class=\\"navlink-custom text-decoration-none text-uppercase text-white\\" href=\\"/contact\\"><img src=\\"/svg/email-light.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-light\\">Contact</span></a></li></ul></div></nav></div>"`;

exports[`navigation menu snapshots, light/dark mode, absolute/relative position 3`] = `"<div class=\\"px-4 px-sm-5 w-100 position-relative\\" style=\\"\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo-dark.png\\" class=\\"logo\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider dark-burger \\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><li><a class=\\"navlink-custom text-decoration-none text-uppercase \\" href=\\"/speedcubing\\"><img src=\\"/svg/cubing-dark.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-dark\\">Speedcubing</span></a></li><li><a class=\\"navlink-custom text-decoration-none text-uppercase \\" href=\\"/contact\\"><img src=\\"/svg/email-dark.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-dark\\">Contact</span></a></li></ul></div></nav></div>"`;

exports[`navigation menu snapshots, light/dark mode, absolute/relative position 4`] = `"<div class=\\"px-4 px-sm-5 w-100 position-absolute\\" style=\\"z-index: 9999;\\"><nav class=\\"navbar navbar-expand-md\\"><a href=\\"/\\"><img alt=\\"logo\\" src=\\"/logos/je-logo-dark.png\\" class=\\"logo\\"></a><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"navbar-toggler\\"><div class=\\"hamburger hamburger--slider dark-burger \\"><div class=\\"hamburger-box\\"><div class=\\"hamburger-inner\\"></div></div></div></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"mr-auto navbar-nav\\"><li><a class=\\"navlink-custom text-decoration-none text-uppercase \\" href=\\"/speedcubing\\"><img src=\\"/svg/cubing-dark.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-dark\\">Speedcubing</span></a></li><li><a class=\\"navlink-custom text-decoration-none text-uppercase \\" href=\\"/contact\\"><img src=\\"/svg/email-dark.svg\\" alt=\\"\\" class=\\"menu-icon\\" width=\\"24\\" height=\\"24\\"><span class=\\"underline-dark\\">Contact</span></a></li></ul></div></nav></div>"`;
